iree/compiler/Codegen/LLVMCPU/VectorContractCustomKernels.cpp

```diff
@@ -150,6 +151,9 @@ static bool isMatrixTimesMatrixTransposedOfGivenShape(
 
 // Checks that the Value `extResult` is defined by an arith::ExtSIOp promoting
 // from `extSrcType` to `extDstType`, and returns the input of the ExtSIOp.
+// Note that this only looks at the immediately defining operation, so we likely
```

Comment:
The pattern matching here only works if you have (pseudo IR)

```mlir
%lhs_wide = arith.extsi %lhs : tensor<8x4xi8> to tensor<8x4xi32>
%rhs_wide = arith.extsi %rhs : tensor<4x8xi8> to tensor<4x8xi32>
%mmt4d = vector.contract {...} %lhs_wide, %rhs_wide... 
```

not if you have 

```mlir
%lhs_wide = arith.extsi %lhs : tensor<8x4xi8> to tensor<8x4xi32>
%lhs_wide_doubled = arith.add %lhs_wide, %lhs_wide : tensor<8x4xi32>
%rhs_wide = arith.extsi %rhs : tensor<4x8xi8> to tensor<4x8xi32>
%mmt4d = vector.contract {...} %lhs_wide_doubled, %rhs_wide... 
```

If something intervenes between the cast and the vector contract the pattern matching gets messed up. We'd want to sink the widening as far down as possible, I think. Of course sinking this past the doubling gets more complicated since we have to deal with overflow. Anyway, that's what this is referring to.